### PR TITLE
FMWK-393 Implement support for single-file backup

### PR DESCRIPTION
**Pull request by filkeith**
- implemented single file backup

### CODE REVIEW

File: config_backup.go [hunk 1]
```diff
@@ -70,6 +70,8 @@ type BackupConfig struct {
 	// File size limit (in bytes) for the backup. If a backup file exceeds this
 	// size threshold, a new file will be created. 0 for no file size limit.
 	FileLimit int64
+	// Backup to a single backup file. If is set, params Parallel FileLimit must not be set.
+	OutputFile string
```
Comment (korotkov-aerospike) [inline config_backup.go:74]: I really don't think that backup config should store it. It's the Writer's resposibity.

File: handler_restore.go [hunk 2]
```diff
@@ -39,6 +39,11 @@ type StreamingReader interface {
 	// Must be run in a goroutine `go rh.reader.StreamFiles(ctx, readersCh, errorsCh)`.
 	StreamFiles(context.Context, chan<- io.ReadCloser, chan<- error)
 
+	// OpenFile creates readers from single file and sends them to the channel.
+	// In case of an error, the error is sent to the error channel.
+	// Must be run in a goroutine `go rh.reader.StreamFiles(ctx, readersCh, errorsCh)`.
+	OpenFile(ctx context.Context, filename string, readersCh chan<- io.ReadCloser, errorsCh chan<- error)
```
Comment (korotkov-aerospike) [inline handler_restore.go:45]: Also, I don't like the new function, It's the StreamingReader's responsibility to provide an io.reader

File: io/local/writer.go [hunk 3]
```diff
@@ -39,26 +43,37 @@ const bufferSize = 4096 * 1024 // 4mb
 // If non-zero, FileSizeLimit must be greater than or equal to 1MB.
 // FileSizeLimit is not a strict limit, the actual file size may exceed this
 // limit by a small amount.
-func NewDirectoryWriterFactory(dir string, removeFiles bool,
+func NewDirectoryWriter(removeFiles bool, opts ...Opts,
```
Comment (korotkov-aerospike) [inline io/local/writer.go:46]: make remove files an opts too?

**Comment by filkeith** [comment]
@reugn re-review

**Comment by filkeith** [comment]
also, there is no races, I cheked

**Review by reugn — APPROVED** [review]

**Pull request merged**